src/vsr/replica.zig

```diff
@@ -711,7 +711,17 @@ pub fn ReplicaType(
                 // that SV to a DVC (dropping the hooks), and never finished the view change.
                 if (op_head == null) {
                     assert(self.view > self.log_view);
-                    op_head = self.journal.op_maximum();
+                    op_head = @max(self.journal.op_maximum(), self.op_checkpoint());
```

Comment:
One thing that I am a _bit_ worried about here though (and this applied before this PR as well). If we *do* end up using `self.journal.op_maximum()` as the `op_head` here, wouldn't we also have to invoke `op_head_certain` for the cases where we rely on the journal? I _think_ this has always been missing. Or am _I_ missing something? 🤔

Reply:
Hmm, I think I'm missing some context here. Are you referring to this case:

```
                    // Don't call op_head_certain() here, as we didn't use the journal to infer our
                    // head op. We used only vsr_headers, and a DVC always has a certain head op.
                    assert(self.view > self.log_view);
                    self.transition_to_view_change_status(self.view);
```

-- i.e. we were in view-change protocol before crashing?

Reply:
Oops, I've been thinking about this particular flows for a bit now so sometimes I rush through the context. 😆 

Yes, that's precisely what I was talking about! We currently don't invoke `op_head_certain` for this case because we assume we're _only_ using the DVC headers and not the journal. But we _do_ have this code where we use the journal even when self.log_view < self.view, right? 
https://github.com/tigerbeetle/tigerbeetle/blob/ed0a1c73a24b2caa9fbcecf54b52dfbb27ded83d/src/vsr/replica.zig#L709-L716

Reply:
Or is it perhaps safe to not call `op_head_certain` since we're anyway jumping into view change and would learn about the head via a start_view when the view change completes? Till then, we're safe since we're reusing our old DVC headers anyway (instead of regenerating them using the journal)? 🤔

Reply:
> Or is it perhaps safe to not call op_head_certain since we're anyway jumping into view change and would learn about the head via a start_view when the view change completes? Till then, we're safe since we're reusing our old DVC headers anyway (instead of regenerating them using the journal)?

I think this is correct -- we are reusing our DVC headers (which originated before our crash) so I think it is safe to skip `op_head_certain`.

Reply:
Thank you for confirming, dj! Resolving this thread.